<a href="https://colab.research.google.com/github/priyansh1997/AlmaBetter/blob/main/Priyansh_Tripathi_EDA_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# 1.1 Importing Libraries
These libraries will be used for Exploratory Data Analysis.

In [69]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

In [70]:
# Removing warnings
# import warnings
# warnings.filterwarnings("ignore")

# 1.2 Data Loading
Loading the dataset from google drive.

In [71]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [72]:
play_store_df=pd.read_csv('/content/drive/MyDrive/datasets/Play Store Data.csv')
user_reviews_df=pd.read_csv('/content/drive/MyDrive/datasets/User Reviews.csv')

# 1.3 Data Inspection
Inspection of the data contains information about the data, head/tail part, description of dataset, shape etc.

In [73]:
play_store_df.head()

App  ...   Android Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [74]:
user_reviews_df.head()

App  ... Sentiment_Subjectivity
0  10 Best Foods for You  ...               0.533333
1  10 Best Foods for You  ...               0.288462
2  10 Best Foods for You  ...                    NaN
3  10 Best Foods for You  ...               0.875000
4  10 Best Foods for You  ...               0.300000

[5 rows x 5 columns]

In [75]:
play_store_df.shape

(10841, 13)

In [76]:
user_reviews_df.shape

(64295, 5)

In [77]:
play_store_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


Here we are having table showing 13 feature names and the count of non-values present in them. Now we will check each column in data cleaning part. 

In [78]:
user_reviews_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 64295 entries, 0 to 64294
Data columns (total 5 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   App                     64295 non-null  object 
 1   Translated_Review       37427 non-null  object 
 2   Sentiment               37432 non-null  object 
 3   Sentiment_Polarity      37432 non-null  float64
 4   Sentiment_Subjectivity  37432 non-null  float64
dtypes: float64(2), object(3)
memory usage: 2.5+ MB


Here also in user reviews dataframe we have 5 features with there counts of non-null values.

In [79]:
play_store_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [80]:
# Checking the column names and removing the spaces so column will be easier to access 
play_store_df.rename(columns={"Content Rating":"Content_Rating", "Last Updated":"Last_Updated", "Current Ver":"Current_Ver", "Android Ver":"Android_Ver"}, inplace=True)

In [81]:
play_store_df.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content_Rating', 'Genres', 'Last_Updated', 'Current_Ver',
       'Android_Ver'],
      dtype='object')

In [82]:
user_reviews_df.columns 

Index(['App', 'Translated_Review', 'Sentiment', 'Sentiment_Polarity',
       'Sentiment_Subjectivity'],
      dtype='object')

Already spaces are replaced by underscores

### 1.3.1 Data Cleaning part for Play store dataframe.

Firstly we will remove the duplicated rows for the play store dataframe.

In [83]:
duplicated_rows_play_store_index=play_store_df[play_store_df.duplicated()].index
print(duplicated_rows_play_store_index)

Int64Index([  229,   236,   239,   256,   261,   265,   266,   267,   268,
              269,
            ...
             8632,  8635,  8636,  8639,  8642,  8643,  8654,  8658, 10049,
            10768],
           dtype='int64', length=483)


In [84]:
play_store_df.drop(duplicated_rows_play_store, inplace=True)
# Dropping the duplicated rows as they were disturbing the data integrity and checking the shape
play_store_df.shape

(10358, 13)

In [85]:
play_store_df.App.isnull().sum()

0

No null values in App feature.

In [86]:
play_store_df.Category.isnull().sum()

0

No null values in Category feature.

In [87]:
# Getting the unique values of category column
play_store_df.Category.unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

These are all the unique categories of application that we are having in our data.

In [88]:
play_store_df.Rating.isnull().sum()

1465

We are having 1465 null values in rating column which we have to fill or drop out.

In [89]:
print("Minimum rating to an App given:",min(play_store_df.Rating))
print("Maximum rating to an App given:",max(play_store_df.Rating))

Minimum rating to an App given: 1.0
Maximum rating to an App given: 19.0


Max rating of the App can be 5.0

In [90]:
play_store_df[play_store_df.Rating>5]

App Category  ...  Current_Ver Android_Ver
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9  ...   4.0 and up         NaN

[1 rows x 13 columns]

We can see this record is manipulated as it has rating more than 5, price is everyone which should be some value, installs is free etc. so we will remove this record.

In [92]:
play_store_df.drop(play_store_df[play_store_df.Rating>5].index,inplace=True)

In [96]:
print("Now maximum rating of an App is:",max(play_store_df.Rating))

Now maximum rating of an App is: 5.0


In [97]:
play_store_df.reset_index(drop=True,inplace=True)

In [98]:
play_store_df.head()

App  ...   Android_Ver
0     Photo Editor & Candy Camera & Grid & ScrapBook  ...  4.0.3 and up
1                                Coloring book moana  ...  4.0.3 and up
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ...  4.0.3 and up
3                              Sketch - Draw & Paint  ...    4.2 and up
4              Pixel Draw - Number Art Coloring Book  ...    4.4 and up

[5 rows x 13 columns]

In [99]:
play_store_df.Rating.isnull().sum()

1465

Still we are having 1465 number of null values in our Rating column.

In [100]:
versions_of_app=play_store_df[play_store_df.App.duplicated(keep=False)].groupby(by=["App"],as_index=False).apply(lambda x: x[:])
versions_of_app

App  ...         Android_Ver
0   2697                            365Scores - Live Scores  ...          4.1 and up
    4959                            365Scores - Live Scores  ...          4.1 and up
1   1468                                        8 Ball Pool  ...        4.0.3 and up
    1496                                        8 Ball Pool  ...        4.0.3 and up
    1544                                        8 Ball Pool  ...        4.0.3 and up
...                                                     ...  ...                 ...
520 2331                        textPlus: Free Text & Calls  ...          4.1 and up
521 2602  theScore: Live Sports Scores, News, Stats & Vi...  ...          4.4 and up
    2680  theScore: Live Sports Scores, News, Stats & Vi...  ...          4.4 and up
522 3226                     wetter.com - Weather and Radar  ...  Varies with device
    7820                     wetter.com - Weather and Radar  ...  Varies with device

[1221 rows x 13 columns]

In [101]:
versions_of_app[versions_of_app["Rating"].isnull()]

App        Category  ...  Current_Ver Android_Ver
48  2222          Blood Pressure         MEDICAL  ...          1.0  2.3 and up
282 23    Mcqueen Coloring pages  ART_AND_DESIGN  ...        1.0.0  4.1 and up
    1878  Mcqueen Coloring pages          FAMILY  ...        1.0.0  4.1 and up

[3 rows x 13 columns]

Checking for the different versions and filling the rating of Applications with there mean.

In [102]:
play_store_df[play_store_df["App"]=='Blood Pressure']

App Category  Rating  ...    Last_Updated Current_Ver Android_Ver
2052  Blood Pressure  MEDICAL     4.2  ...   July 24, 2018      3.27.3  4.1 and up
2222  Blood Pressure  MEDICAL     NaN  ...  March 14, 2015         1.0  2.3 and up

[2 rows x 13 columns]

Only 2 versions available so we can fill with the mean of category to which this app belongs.

In [103]:
play_store_df[play_store_df["App"]=='Mcqueen Coloring pages']

App        Category  ...  Current_Ver Android_Ver
23    Mcqueen Coloring pages  ART_AND_DESIGN  ...        1.0.0  4.1 and up
1878  Mcqueen Coloring pages          FAMILY  ...        1.0.0  4.1 and up

[2 rows x 13 columns]

Here also only 2 applications are there and they are also present in the different category so we will put catgory wise mean in both.

In rest of the records with null-values in rating column we will fill category wise mean because we don't have different versions of those applications.

In [104]:
play_store_df['Rating'].fillna(play_store_df.groupby('Category')['Rating'].transform('mean'), inplace=True)

In [105]:
play_store_df['Reviews']=play_store_df['Reviews'].astype('int')